### PR TITLE
feat: [DEL-3274]: migration to create default ng delegate tokens for …

### PR DESCRIPTION
**Pull request by arpit3107**
…account,org and project.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30565)

### CODE REVIEW

**Comment by abhinav-harness** [comment]
___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 56 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDusw7-h3BF8eKfdua:-MvDusw7-h3BF8eKfdub:b-nuf0ho) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L56)):*
> ```Java
>             .forEach(projectIdentifier
>                 -> ownerList.add(DelegateEntityOwnerHelper.buildOwner(orgIdentifier, projectIdentifier)));
>       }
> ```

add one log here for collected owners

**Comment by abhinav-harness** [comment]
___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 57 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDuwqV-0VnLQGkHY3l:-MvDuwqV-0VnLQGkHY3m:b-kfkn4l) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L57)):*
> ```Java
>                 -> ownerList.add(DelegateEntityOwnerHelper.buildOwner(orgIdentifier, projectIdentifier)));
>       }
>       ownerList.forEach(owner -> delegateNgTokenService.upsertDefaultToken(accountId, owner, true));
> ```

not idempotent

**Comment by abhinav-harness** [comment]
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 46 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDv6Nw56k1hgpBAPUY:-MvDv6Nw56k1hgpBAPUZ:b8fd19t) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L46)):*
> ```Java
>     })
> @NextGenManagerAuth
> @Hidden
> ```

this should not be in 930 module

**Comment by abhinav-harness** [comment]
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 86 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDvAWo5T5ccSgywFHF:-MvDvAWo5T5ccSgywFHG:bvq28ne) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L86)):*
> ```Java
>     List<String> projectIdentifiers = projectService
>                                           .list(Criteria.where(ProjectKeys.accountIdentifier)
>                                                     .is(accountIdentifier)
> ```

dont we have project list and org list client already?

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30565)* status: 0 of 1 approvals obtained.

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30565)* status: 3 file(s) to review. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 56 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDusw7-h3BF8eKfdua:-MvEUQxe6uEZNkfv0BAv:b-z5dvmh) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L56)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

add one log here for collected owners
</blockquote></details>

Didnt got exactly?
___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 57 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDuwqV-0VnLQGkHY3l:-MvEUV2oAqtCOXcy4wbA:bqqgvxu) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L57)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

not idempotent
</blockquote></details>

Yes, it is idempotent. The last parameter of { delegateNgTokenService.upsertDefaultToken(accountId, owner, true  } is true, which means skips creating a new token if its already there.
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 46 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDv6Nw56k1hgpBAPUY:-MvEUsBt2y2LuUh3nPvY:bgoqneg) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L46)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

this should not be in 930 module
</blockquote></details>

I agree, but I cant find anyother better place. For my testing, i have moved it to 930-ng-license-manager. Suggest me a better module, (may be 950-ng-project-n-orgs)
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 86 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDvAWo5T5ccSgywFHF:-MvEVIEoBwhoVoJec2ZX:bp7iv56) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L86)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

dont we have project list and org list client already?
</blockquote></details>

yes we have, but that is on 120-ng-manager, and it cant be moved below 390-db-migrations module.

File: 930-ng-license-manager/BUILD.bazel [hunk 1]
```diff
@@ -13,6 +13,7 @@ java_library(
         "//935-subscription-sdk:module",
         "//945-account-mgmt:module",
         "//948-access-control-sdk:module",
+        "//950-ng-project-n-orgs:module",
```
Comment (arpit3107) [inline 930-ng-license-manager/BUILD.bazel:16]: This is not the final destination, just for my local testing.

**Comment by abhinav-harness** [comment]
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 86 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDvAWo5T5ccSgywFHF:-MvEfeuV0nxSh6A8G_oi:b-qr4lz4) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L86)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

yes we have, but that is on 120-ng-manager, and it cant be moved below 390-db-migrations module.
</blockquote></details>

i think we should have something in ng core clients module

**Comment by abhinav-harness** [comment]
___
*[930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java, line 46 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDv6Nw56k1hgpBAPUY:-MvEflT--prjWYLFaqsO:b-4u5zn1) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/930-ng-core-clients/src/main/java/io/harness/delegate/DelegateNgTokenMigrationResource.java#L46)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

I agree, but I cant find anyother better place. For my testing, i have moved it to 930-ng-license-manager. Suggest me a better module, (may be 950-ng-project-n-orgs)
</blockquote></details>

lets have a call tomorrow

**Comment by abhinav-harness** [comment]
___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 56 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDusw7-h3BF8eKfdua:-MvEftJ10v-j4zalUsx0:b-oa8wl1) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L56)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Didnt got exactly?
</blockquote></details>

add log line about owners where you are going to run this

**Comment by abhinav-harness** [comment]
___
*[390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java, line 57 at r1](https://reviewable.io/reviews/harness/harness-core/30565#-MvDuwqV-0VnLQGkHY3l:-MvEg5cH7rBo0wJ65eqJ:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/ef736f5e412a1898e41d7b0aa10f2191e60bdce1/390-db-migration/src/main/java/io/harness/migrations/all/CreateDefaultDelegateNgTokenMigration.java#L57)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Yes, it is idempotent. The last parameter of { delegateNgTokenService.upsertDefaultToken(accountId, owner, true  } is true, which means skips creating a new token if its already there.
</blockquote></details>

ok

**Pull request closed without merging**